app/Repositories/Activity/ActivityRepository.php

```diff
@@ -6,16 +6,190 @@
+                    if (!isset($projects[$activityPlaylistProjectid])) {
+                        $projects[$activityPlaylistProjectid] = $activityPlaylist->project->attributesToArray();
+                        $projects[$activityPlaylistProjectid]['playlists'] = [];
+                    }
+
+                    if (!isset($projects[$activityPlaylistProjectid]['playlists'][$activityPlaylistId])) {
+                        $projects[$activityPlaylistProjectid]['playlists'][$activityPlaylistId] = $activityPlaylist->attributesToArray();
+                        $projects[$activityPlaylistProjectid]['playlists'][$activityPlaylistId]['activities'] = [];
+                    }
+
+                    $activityModel = $searchedModel->attributesToArray();
+                    $projects[$activityPlaylistProjectid]['playlists'][$activityPlaylistId]['activities'][$activityId] = $activityModel;
+                }
+            }
+        }
+
+        return $projects;
+    }
+
+    /**
+     * Get the advance search request
+     *
+     * @param  Request  $request
+     * @return Collection
+     */
+    public function advanceSearchForm($request)
+    {
+        $counts = [];
+        $projectIds = [];
+
+        if ($request->has('userIds') && $request->filled('userIds')) {
+            $userIds = $request->input('userIds');
+
+            $projectIds = Project::whereHas('users', function (Builder $query) use($userIds) {
+                $query->whereIn('id', $userIds);
+            })->pluck('id')->toArray();
+        }
+
+        $searchResultQuery = Activity::searchForm()
+                            ->query($request->input('query', 0))
+                            ->join(Project::class, Playlist::class)
+                            ->aggregate('count_by_index', [
+                                'terms' => [
+                                    'field' => '_index',
+                                ]
+                            ])
+                            ->shared(true)
+                            ->type($request->input('type', 0))
+                            ->subjectIds($request->input('subjectIds', []))
+                            ->educationLevelIds($request->input('educationLevelIds', []))
+                            ->projectIds($projectIds)
+                            ->negativeQuery($request->input('negativeQuery', 0))
+                            ->sort($request->input('sort', '_id'), $request->input('order', 'desc'))
+                            ->from($request->input('from', 0))
+                            ->size($request->input('size', 10));
+
+        if ($request->has('model') && $request->filled('model')) {
+            $searchResultQuery = $searchResultQuery->postFilter('term', ['_index' => $request->input('model')]);
+        }
+
+        $searchResult = $searchResultQuery->execute();
+
+        $aggregations = $searchResult->aggregations();
```

Comment:
aggregations?

Reply:
Its function from the Elasticsearch package we are using.

https://github.com/babenkoivan/elastic-scout-driver-plus#aggregate